src/huggingface_hub/keras_mixin.py

```diff
@@ -16,6 +16,120 @@
 
 logger = logging.getLogger(__name__)
 
+if is_tf_available():
+    import tensorflow as tf
+
+
+def _extract_hyperparameters_from_keras(model):
+    hyperparameters = dict()
+    if hasattr(model, "optimizer") and model.optimizer is not None:
+        hyperparameters["optimizer"] = model.optimizer.get_config()
+    else:
+        hyperparameters["optimizer"] = None
+    hyperparameters[
+        "training_precision"
+    ] = tf.keras.mixed_precision.global_policy().name
+
+    return hyperparameters
+
+
+def _parse_model_history(model):
+    logs = model.history.history
+    num_epochs = len(logs["loss"])
+    lines = []
+
+    for value in range(num_epochs):
+        epoch_dict = {
+            log_key: log_value_list[value] for log_key, log_value_list in logs.items()
+        }
+        values = dict()
+        for k, v in epoch_dict.items():
+            if k.startswith("val_"):
+                k = "validation_" + k[4:]
+            elif k != "epoch":
+                k = "train_" + k
+            splits = k.split("_")
+            name = " ".join([part.capitalize() for part in splits])
+            values[name] = v
+        lines.append(values)
+    return lines
+
+
+def _plot_network(model, save_directory):
+    tf.keras.utils.plot_model(
```

Comment:
Nit: as discussed let's try to use `svg` if possible

Reply:
it doesn't work :(

Reply:
plot_model() doesn't accept svg.